source/concepts.tex

```diff
@@ -0,0 +1,1310 @@
+  template<class T>
+  concept Swappable = @\seebelow@;
+
+  template<class T, class U>
+  concept SwappableWith = @\seebelow@;
+
+  // \ref{concept.destructible}, Destructible
+  template<class T>
+  concept Destructible = @\seebelow@;
+
+  // \ref{concept.constructible}, Constructible
+  template<class T, class... Args>
+  concept Constructible = @\seebelow@;
+
+  // \ref{concept.defaultconstructible}, DefaultConstructible
+  template<class T>
+  concept DefaultConstructible = @\seebelow@;
+
+  // \ref{concept.moveconstructible}, MoveConstructible
+  template<class T>
+  concept MoveConstructible = @\seebelow@;
+
+  // \ref{concept.copyconstructible}, CopyConstructible
+  template<class T>
+  concept CopyConstructible = @\seebelow@;
+
+  // \ref{concepts.compare}, comparison concepts
+  // \ref{concept.boolean}, Boolean
+  template<class B>
+  concept Boolean = @\seebelow@;
+
+  // \ref{concept.equalitycomparable}, EqualityComparable
+  template<class T>
+  concept EqualityComparable = @\seebelow@;
+
+  template<class T, class U>
+  concept EqualityComparableWith = @\seebelow@;
+
+  // \ref{concept.stricttotallyordered}, StrictTotallyOrdered
+  template<class T>
+  concept StrictTotallyOrdered = @\seebelow@;
+
+  template<class T, class U>
+  concept StrictTotallyOrderedWith = @\seebelow@;
+
+  // \ref{concepts.object}, object concepts
+  // \ref{concept.movable}, Movable
+  template<class T>
+  concept Movable = @\seebelow@;
+
+  // \ref{concept.copyable}, Copyable
+  template<class T>
+  concept Copyable = @\seebelow@;
+
+  // \ref{concept.semiregular}, Semiregular
+  template<class T>
+  concept Semiregular = @\seebelow@;
+
+  // \ref{concept.regular}, Regular
+  template<class T>
+  concept Regular = @\seebelow@;
+
+  // \ref{concepts.callable}, callable concepts
+  // \ref{concept.invocable}, Invocable
+  template<class F, class... Args>
+  concept Invocable = @\seebelow@;
+
+  // \ref{concept.regularinvocable}, RegularInvocable
+  template<class F, class... Args>
+  concept RegularInvocable = @\seebelow@;
+
+  // \ref{concept.predicate}, Predicate
+  template<class F, class... Args>
+  concept Predicate = @\seebelow@;
+
+  // \ref{concept.relation}, Relation
+  template<class R, class T, class U>
+  concept Relation = @\seebelow@;
+
+  // \ref{concept.strictweakorder}, StrictWeakOrder
+  template<class R, class T, class U>
+  concept StrictWeakOrder = @\seebelow@;
+}
+\end{codeblock}
+
+\rSec1[concepts.lang]{Language-related concepts}
+
+\rSec2[concepts.lang.general]{General}
+
+\pnum
+This section contains the definition of concepts corresponding to language
+features. These concepts express relationships between types, type
+classifications, and fundamental type properties.
+
+\pnum
+\begin{example}
+Several concept specifications in this subclause are specified in terms of a
+type trait, and include an annotation that there need be no subsumption
+relationship between the concept and the type trait. Given the following set of
```

Comment:
This doesn't make sense. There *cannot* be any subsumption relationship between a concept and an arbitrary non-concept expression. The core language doesn't allow it.

We should strike this paragraph and all subsequent "There need be no subsumption relationship" paragraphs.

Reply:
If you like, you could convert this example to a note that there will be no such subsumption relationship.

Reply:
Yes, that wording is intended to convey that we mean the same utterance of that expression. Please feel free to file an editorial issue to clarify that.

Reply:
I've made this change, and pushed another commit to merge `Integral`, `SignedIntegral`, and `UnsignedIntegral` into a single subclause - they are now too small for individual clauses.